src/ThoralfPlugin/Encode/Bool.hs

```diff
@@ -110,7 +115,19 @@ compLitMaker :: Vec Two String -> Vec 'Zero String -> String
 compLitMaker (x :> y :> VNil) VNil =
   "(or (< " ++ x ++ " " ++ y ++ ")  (= " ++ x ++ " " ++ y ++ "))"
 
+ifConv :: TyCon -> Type -> Maybe TyConvCont
+ifConv ifCon ty = do
+  (tycon, types) <- splitTyConApp_maybe ty
+  case (tycon == ifCon, types) of
+    -- N.B. The first argument is the kind
+    (True, (_ : x : y : z : xs)) -> return $
```

Comment:
In general I'm a bit suspicious of matching on a wildcard tail (e.g. `xs`) and then throwing it away but this is what is done elsewhere. Is there a reason why this is necessary?

Reply:
It seems like the safe thing to do would be to rather match on a `[]`. I'll do this.